Common/TableProducer/PID/pidTPC.cxx

```diff
@@ -170,36 +199,50 @@ struct tpcPid {
     if (useNetworkCorrection) {
 
       auto start_overhead = std::chrono::high_resolution_clock::now();
-      std::vector<float> track_properties;
+
+      // Defining some network parameters
+      int inputDimensions = network.getInputDimensions();
+      int outputDimensions = network.getOutputDimensions();
+      const unsigned long track_prop_size = tracks.size() * 9;
```

Comment:
This is for every particle species, right? Would it require less space if one was interested in e.g. only pions and kaons?

Reply:
Indeed that would be the case, however the correction is performed per track and since we don't know a priori which track has which species associated with it, we would need to produce the correction for each mass hypotheses. If we consider that the analyser might already trust the initial Bethe-Bloch parametrisation and assigns the mass hypotheses based on the distance to the nearest BB-curve, then indeed, we could save a lot of memory! The question is whether we want to make assumptions already before the correction is applied (this might lead to misidentification (on a small scale I guess), but would save a lot of memory of course). If this hypotheses would be used, then we would only calculate for each track a single mass hypotheses, thus reducing memory consumption by a factor 9.